src/NuGet.Core/NuGet.Build.Tasks.Pack/NuGet.Build.Tasks.Pack.targets

```diff
@@ -452,6 +462,7 @@ Copyright (c) .NET Foundation. All rights reserved.
     <!-- The below workaround needs to be done due to msbuild bug https://github.com/Microsoft/msbuild/issues/3121 -->
     <ItemGroup>
       <TfmSpecificPackageFileWithRecursiveDir Include="@(TfmSpecificPackageFile)">
+        <NuGetTfm>$(TargetFramework)</NuGetTfm>
```

Comment:
naming: Should we call it TargetFramework to match BuildOutputInPackage?
https://github.com/NuGet/NuGet.Client/blob/1466fd7767c92b84fc4c22827ea9c3bf0ec82f5b/src/NuGet.Core/NuGet.Build.Tasks.Pack/NuGet.Build.Tasks.Pack.targets#L66-L70

Alternatively, I wonder if an ItemDefinitionGroup solves our problem faster, similar to the way BuildOutputInPackage does it.